### PR TITLE
Use new ESRI geocoder endpoint

### LINKED ISSUE #688: Update geocoder service
`find` endpoint has been deprecated: https://developers.arcgis.com/rest/geocode/api-reference/whats-new-world-geocoding-service.htm

Also investigate if ranking by distance works with a bounding box now. (Currently do not rank results by distance, only limit to bounding box.)

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Replace deprecated `find` endpoint with new `findAddressCandidates` endpoint.
New endpoint returns information in a different structure.
New endpoint now supports using both search bounding box and bias location,
so add logic to bias typeahead suggestions to favor by proximity to current map center.

Closes #688.

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
@lederer with this, the first address autocomplete suggestion should become the likeliest.

**Comment by flibbertigibbet** [comment]
Geocoder endpoint changelog: https://developers.arcgis.com/rest/geocode/api-reference/whats-new-world-geocoding-service.htm

File: src/app/scripts/cac/search/cac-geocoder.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 CAC.Search.Geocoder = (function ($, SearchParams) {
     'use strict';
 
-    var url = 'https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/find';
+    var url = 'https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/findAddressCandidates';
     var reverseUrl = 'https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode';
 
     var defaults = {
-        bbox: SearchParams.searchBounds,
+        searchExtent: SearchParams.searchBounds,
```
Comment (KlaasH) [inline src/app/scripts/cac/search/cac-geocoder.js]: The one in `SearchParams` was also renamed.

File: src/app/scripts/cac/search/cac-typeahead.js [hunk 2]
```diff
@@ -182,7 +182,16 @@ CAC.Search.Typeahead = (function (_, $, Geocoder, SearchParams, Utils) {
                     // do not behave well with the follow-up geocode.
                     // See: https://github.com/azavea/cac-tripplanner/issues/307
                     return _.filter(list.suggestions, { isCollection: false });
-                }
+                },
+                replace: function(url, query) {
```
Comment (KlaasH) [inline src/app/scripts/cac/search/cac-typeahead.js]: Couldn't this be done by just adding the two params to the `url` above? e.g.
```
url: url + '?text=%QUERY&' + $.param(_.extend({}, params, {
   location: SearchParams.getLocation(), distance: SearchParams.distance}))
```

**Comment by flibbertigibbet** [comment]
Okay, think this should be good to go now.

**Pull request merged**